### DISCUSSION TITLE
Linking to specific headings and paragraphs!!!

### DISCUSSION THREAD

**Discussion by renerocksai** [nvim-telekasten/telekasten.nvim#14] [Announcements]
Whaaaat a piece of work! Phew! :sweat_smile: 

Live from the [README](https://github.com/renerocksai/telekasten.nvim/blob/main/README.md):

Telekasten now supports linking to headings and paragraphs within notes. The following link types are supported:

```markdown
# Note links
- [[A cool title]]  ................. links to the note named 'A cool title'
- [[A cool title#Heading 27]]  ...... links to the heading 'Heading 27' within the note 
                                      named 'A cool title'
- [[A cool title#^xxxxxxxx]]  ....... links to the paragraph with id ^xxxxxxxx within the note 
                                      named 'A cool title'
- [[#Heading 27]]  .................. links to the heading 'Heading 27' within all notes
- [[#^xxxxxxxx]]  ................... links to the paragraph with id ^xxxxxxxx within all notes

## Optionally, notes can live in specific sub-directories
- [[some/subdirectory/A cool title]]  ................. links to note named 'A cool title' 
                                                        in some/subdirectory
- [[some/subdirectory/A cool title#Heading 27]]  ...... links to the heading 'Heading 27' within 
                                                        the note named 'A cool title' 
                                                        in some/subdirectory
- [[some/subdirectory/A cool title#^xxxxxxxx]]  ....... links to the paragraph with 
                                                        id ^xxxxxxxx within the note named 
                                                        'A cool title' in some/subdirectory


# Media links
Use these for images, PDF files, videos. If telescope-media-files is installed, these can 
be previewed.
- ![optional title](path/to/file) ... links to the file `path/to/file`
```
You can `follow_link()` all of the above, with the exception of media links, which you can `preview_img()`.

Note that notes linked to with headings or paragraph IDs **will not be created automatically**. Non-existing notes will
be ignored, a global search will be performed instead.

For now, you have to create such links yourself.  You can edit existing links and append headings, for example: From `[[The note]]` to `[[The note#Some heading or subheading]]`.

Regarding linking to paragraphs: The `^blockid` notation is supported by more and more tools now. It works like this:

- at the end of a line / paragraph you want to link to, you put some block id marker in the form of `^block-id`.
- typically, block ids are tool-generated - but no one can stop you defining your own.
    - tool-generated block ids look like this: `^xAcSh-xxr`. 
- instead of linking to a heading, you use the block id as heading: `[[Some note#^block-id]]`.
- instead of putting the block id at the end of a line, you can also make a line break and put it in the next line, like
  illustrated below.

  ```markdown
  Here we have a line or even a paragraph. 
  We don't want to have a block id dangling
  at the end of a line. So instead, we just
  put it in the next one.
  ^xxxblkxx

  Here goes the next paragraph.
  ```

**Comment by lokesh-krishna** [comment]
I was going to request this but felt like it would be asking too much and then I see this. You, sir, are a hero.

**Comment by kraxli** [comment]
that is very nice :-) - thanks for all of it!

Just wondering how I can insert a link to a header ```[[file_name#header title]]```?  
```
lua require("telekasten").insert_link({ i=false/true })
```
does not show headers to insert for me.

**Reply by lambtho12** [reply]
This is not possible yet. 
But if you look for "headings" in the repo issues or discussions, you will see that lots of stuff are discussed on the topic already... 

See this issue for instance, where I explain what I plan to do whenever I have time https://github.com/renerocksai/telekasten.nvim/issues/213#issuecomment-1415319506
